src/test/java/seedu/address/model/person/ModuleNameTest.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.INVALID_MODNAME_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_MODNAME_AMY;
+
+import org.junit.jupiter.api.Test;
+
+public class ModuleNameTest {
+
+    @Test
+    public void toString_invalidModuleName_throwsException() {
+        assertThrows(IllegalArgumentException.class, () -> new ModuleName(INVALID_MODNAME_DESC));
+    }
+
+    @Test
+    public void toString_validModuleName_returnsCorrectString() {
+        assertEquals(VALID_MODNAME_AMY, new ModuleName(VALID_MODNAME_AMY).toString());
+    }
+
+    @Test
+    public void emptyModuleNameTest() {
+        assertEquals(new ModuleName(""), new ModuleName(""));
+    }
+
+    @Test
+    public void equals() {
```

Comment:
good that test case covers various scenarios to reduce bugs in future usage of the field